.semaphore/multi-arch-packaging.yml

```diff
@@ -16,169 +16,55 @@ queue:
-              gh auth status
-            } else {
-                Write-Error "Token file not found at $HOME/.gh-token"
-                exit 1
-            }
-          - npm ci --include=dev
+          - make install-dependencies
       jobs:
-        - name: "Package VSIX"
+        - name: "Package VSIX Files"
+          matrix:
+            - env_var: TARGET
+              values:
+                - darwin-x64
+                - darwin-arm64
+                - linux-x64
+                - linux-arm64
+                - win32-x64
           commands:
-            # Download ide-sidecar third party notices to include in the VSIX
             - |
-              $IDE_SIDECAR_VERSION = Get-Content .versions/ide-sidecar.txt
-              gh release download $IDE_SIDECAR_VERSION --repo confluentinc/ide-sidecar --pattern=THIRD_PARTY_NOTICES.txt --output THIRD_PARTY_NOTICES_IDE_SIDECAR.txt --clobber
-            # Build the VSIX
+              case "$TARGET" in
+                darwin-x64)  export SIDECAR_OS_ARCH=macos-amd64 ;;
+                darwin-arm64) export SIDECAR_OS_ARCH=macos-arm64 ;;
+                linux-x64)   export SIDECAR_OS_ARCH=linux-amd64 ;;
+                linux-arm64) export SIDECAR_OS_ARCH=linux-arm64 ;;
+                win32-x64)   export SIDECAR_OS_ARCH=windows-x64 ;;
+                *) echo "Unknown TARGET: $TARGET" && exit 1 ;;
+              esac
+            - make download-sidecar-executable
+            - make download-third-party-notices-sidecar || true
             - npx gulp bundle
-      epilogue:
-        always:
-          commands:
-            - make ci-bin-sem-cache-store
-        on_pass:
-          commands:
-            # Single multi-line command because Powershell won't track $vsix between commands
-            - |
-              $vsix = Get-ChildItem out -Filter "*.vsix" | Select-Object -ExpandProperty FullName;
-              artifact push workflow $vsix --destination "packaged-vsix-files/$($vsix | Split-Path -Leaf)"
+            - VSIX_FILE=$(find out/ -name "*.vsix")
+            - artifact push workflow ${VSIX_FILE} --destination packaged-vsix-files/$(basename ${VSIX_FILE})
 
   - name: "Upload VSIX files to GitHub"
     run:
       when: "branch =~ '.*' and change_in('/release.svg', {pipeline_file: 'ignore', branch_range: '$SEMAPHORE_GIT_COMMIT_RANGE', default_branch: 'main'})"
     dependencies:
-      - "Package VSIX (MacOS AMD64)"
-      - "Package VSIX (MacOS ARM64)"
-      - "Package VSIX (Linux AMD64)"
-      - "Package VSIX (Linux ARM64)"
-      - "Package VSIX (Windows AMD64)"
+      - "Package VSIX Files"
     task:
-      prologue:
-        commands:
-          - checkout
-          - . vault-setup
-          - artifact pull workflow packaged-vsix-files/
       jobs:
         - name: "Upload VSIX files to GitHub"
           commands:
+            - artifact pull workflow packaged-vsix-files/
             - make upload-vsix-files-to-gh-releases
```

Comment:
TODO: see if we can merge this and https://github.com/confluentinc/vscode/blob/main/.semaphore/multi-arch-packaging.yml together